app/services/bulk_create_test_applications.rb

```diff
@@ -0,0 +1,69 @@
+class BulkCreateTestApplications
+  attr_reader :template_application_form, :candidate, :application_choice
+
+  def initialize(template_application_form)
+    @template_application_form = template_application_form
+    @candidate = template_application_form.candidate
+    @application_choice = template_application_form.application_choices.first
+  end
+
+  def call
+    random_id = SecureRandom.hex(10)
+
+    candidate_field_names_sql, candidate_values_to_persist = compose_sql_fragments(
+      candidate,
+      ignore_fields: %w[id magic_link_token magic_link_token_sent_at],
+      custom_fields: { 'email_address' => "#{random_id}@example.com" },
+    )
+    application_form_field_names_sql, application_form_values_to_persist = compose_sql_fragments(
+      template_application_form,
+      ignore_fields: %w[id candidate_id],
+      custom_fields: { 'last_name' => random_id },
+    )
+    application_choice_field_names_sql, application_choice_values_to_persist = compose_sql_fragments(
+      application_choice,
+      ignore_fields: %w[id application_form_id],
+    )
+
+    query = <<~SQL
+      WITH created_candidate AS (
+        INSERT INTO "candidates" ( #{candidate_field_names_sql})
+        VALUES ( #{candidate_values_to_persist})
+        RETURNING "id"
+      ),
+       created_application_form AS (
+        INSERT INTO "application_forms" ( #{application_form_field_names_sql}, candidate_id )
+        VALUES ( #{application_form_values_to_persist}, (SELECT id FROM created_candidate) )
+        RETURNING "id"
+      )
+
+      INSERT INTO "application_choices" ( #{application_choice_field_names_sql}, application_form_id )
```

Comment:
If we clone an application that is awaiting a provider decision, will cloning the application choices cause the provider users to receive 100k+ emails when they are rejected by default or similar?

Reply:
Good point, yes I think we will. Could include a guard clause protecting against certain application states? Are there any other states we'd want to treat the same way?

Reply:
Guard clause so it only applies to `form started but unsubmitted` sounds like the way to go :+1: and throw a helpful error message if we use it on another type of application

Reply:
Done, ready for another review.